Spanner/src/Database.php

```diff
@@ -427,6 +427,41 @@ public function create(array $options = [])
         return $this->resumeOperation($operation['name'], $operation);
     }
 
+    /**
+     * Update an existing Cloud Spanner database.
+     *
+     * Example:
+     * ```
+     * $operation = $database->updateDatabase(['enableDropProtection' => true]);
+     * ```
+     *
+     * @codingStandardsIgnoreStart
+     * @see https://cloud.google.com/spanner/reference/rpc/google.spanner.admin.database.v1#updatedatabaserequest UpdateDatabaseRequest
+     * @codingStandardsIgnoreEnd
+     *
+     * @param array $options [optional] {
+     *     Configuration Options
+     *
+     *     @type bool $enableDropProtection If true, delete operations for Database
+     *           and Instance will be blocked. **Defaults to** `false`.
+     * }
+     * @return LongRunningOperation<Database>
+     */
+    public function updateDatabase(array $options = [])
+    {
+        return $this->info = $this->connection->updateDatabase([
+            'database' => [
+                'name' => $this->name,
+                'enableDropProtection' => isset($options['enableDropProtection'])
+                    ? $options['enableDropProtection']
+                    : false,
+            ],
+            'updateMask' => [
+                'paths' => ['enable_drop_protection']
```

Comment:
IMO, paths should be set based on whether `enable_drop_protection` field is passed in by user in options. If not, we should keep it empty as we would not want to update the field in that case.

Reply:
Currently, only `enable_drop_protection` field is available in `updateDatabase` RPC. So, if we make this optional to set then the call itself will fail.
So, till we don't have any other flag, we are just setting it to false (if not provided) and adding in `updateMask -> paths`

Reply:
I think it is okay if the updateDatabase RPC fails, because the user is expected to pass in a field. Setting the field to false without the user's intention to set it (especially for fields like drop database protection where the risks are even higher) will be a bigger problem.

Let me know what you think about this.

Reply:
makes sense, I will fix this.